packages/mobile/src/identity/feelessVerification.ts

```diff
@@ -0,0 +1,1061 @@
+  feelessAcceptedAttestationCodesSelector,
+  feelessAttestationCodesSelector,
+  FeelessVerificationState,
+  feelessVerificationStateSelector,
+  isFeelessVerificationStateExpiredSelector,
+} from 'src/identity/reducer'
+import { startAutoSmsRetrieval } from 'src/identity/smsRetrieval'
+import { VerificationStatus } from 'src/identity/types'
+import {
+  AttestationCode,
+  attestationCodeReceiver,
+  completeAttestations,
+  getActionableAttestations,
+  getAttestationsStatus,
+  NUM_ATTESTATIONS_REQUIRED,
+  requestAndRetrieveAttestations,
+  revealAttestations,
+  VERIFICATION_TIMEOUT,
+} from 'src/identity/verification'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import Logger from 'src/utils/Logger'
+import { setMtwAddress } from 'src/web3/actions'
+import { getContractKit } from 'src/web3/contracts'
+import { registerWalletAndDekViaKomenci } from 'src/web3/dataEncryptionKey'
+import { getConnectedAccount, getConnectedUnlockedAccount } from 'src/web3/saga'
+import { TransactionReceipt } from 'web3-eth'
+
+const TAG = 'identity/feelessVerification'
+
+const KOMENCI_URL = 'https://komenci.celo-networks-dev.org'
+// const KOMENCI_URL = 'http://192.168.86.33:3000'
+// TODO: Populate this with expected implementation address
+const ALLOWED_MTW_IMPLEMENTATIONS: Address[] = ['0x88a2b9B8387A1823D821E406b4e951337fa1D46D']
+const CURRENT_MTW_IMPLEMENTATION_ADDRESS: Address = '0x88a2b9B8387A1823D821E406b4e951337fa1D46D'
+
+export function* feelessFetchVerificationState() {
+  Logger.debug(TAG, '@feelessFetchVerificationState', 'Starting fetch')
+  try {
+    ValoraAnalytics.track(VerificationEvents.verification_fetch_status_start)
+    const [contractKit, walletAddress, feelessVerificationState, e164Number]: [
+      ContractKit,
+      string,
+      FeelessVerificationState,
+      string
+    ] = yield all([
+      call(getContractKit),
+      call(getConnectedUnlockedAccount),
+      select(feelessVerificationStateSelector),
+      select(e164NumberSelector),
+      put(feelessSetVerificationStatus(VerificationStatus.GettingStatus)),
+    ])
+
+    const komenciKit = new KomenciKit(contractKit, walletAddress, {
+      url: KOMENCI_URL,
+      token: feelessVerificationState.komenci.sessionToken,
+    })
+
+    try {
+      // Checks if a MTW was verified in a previous attempt and updates state.
+      // This check will likely fail because the pepper hasn't been cached yet
+      // but needs to happen for the edge case that a user has a cached pepper
+      // and Komenci is down
+      yield call(fetchVerifiedMtw, contractKit, walletAddress, e164Number)
+    } catch (error) {
+      Logger.debug(TAG, 'Unable to check it MTW is verified on first attempt')
+    }
+    // Throws error if Komenci is not ready for user, otherwise it updates state
+    yield call(fetchKomenciReadiness, komenciKit)
+
+    // Throws error if unable to retreive phone hash, otherwise it updates state
```

Comment:
> IIUC this throws if the pepper isn't cached? That seems like the wrong behavior if we're going to fetch it from Komenci later

Why does it seem like the wrong behavior? This is essentially just a a helper function that checks if a user already has their own pepper cached. If they don't, and therefore don't know their own phone hash, we bail on the rest of the checks because we don't have the identifier needed to update any progress the user may or may not have.

> In fact, I don't understand why we need to do this at all? Why don't we cache it in the feeless verification state? If it's cached, we'll have it, if not, we won't. I don't see why we need to fetch it here

What if the user has their own pepper already from attempting a non-feeless verification? Why waste time and $$$ asking Komenci for something the user may already have?